From 1c75b49dad2fd664f5a918e9399c0f3710141246 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Tue, 11 Dec 2018 23:52:58 +0200
Subject: [PATCH] [SW-0]: Support vUART to HOST

Using couple registers for transmition (data, status).
Registers are accessiable accross KMD remove and insertion.
This is on top and not instaed of standard UART

Change-Id: I48e4352eb5908cdf231e0394a558ebb7ac13a868
---

diff --git a/preboot.c b/preboot.c
index 387c667..f9b6635 100644
--- a/preboot.c
+++ b/preboot.c
@@ -59,6 +59,21 @@
 #define PREBOOT_VER_STR	"Preboot version " PREBOOT_VER \
 	" (" PREBOOT_DATE " - " PREBOOT_TIME ")"
 
+#ifdef DEV_IS_GOYA
+static void preboot_putc_virt_uart(const char c)
+{
+	/*
+	 * update vUART to HOST
+	 * DATA: mmPCIE_DB_RSV_RSV_0
+	 * STATUS: mmPCIE_DB_RSV_RSV_1
+	 */
+	writel(c, mmPCIE_DB_RSV_RSV_0);
+	writel(1, mmPCIE_DB_RSV_RSV_1);
+}
+#else
+static inline void preboot_putc_virt_uart(const char c) {}
+#endif
+
 static void preboot_puts_raw(const char* str)
 {
 	int i;
@@ -66,6 +81,7 @@
 	for (i=0; str[i] && (i < 128); i++) {
 		while (readl(mmPSOC_UART_0_STATUS) & UART_SR_TXFULL) ;
 		writel(str[i], mmPSOC_UART_0_DATA);
+		preboot_putc_virt_uart(str[i]);
 	}
 }
 
